packages/cli/src/commands/deploy/baremetal.js

```diff
@@ -215,41 +216,81 @@ const commands = (yargs, ssh) => {
     // start/restart monitoring processes
     for (const process of serverConfig.processNames) {
       if (yargs.firstRun) {
-        tasks.push({
-          title: `Starting ${process} process for the first time...`,
-          task: async (_ctx, task) => {
-            await sshExec(ssh, sshOptions, task, serverConfig.path, 'yarn', [
-              'pm2',
-              'start',
-              'ecosystem.config.js',
-              '--only',
-              process,
-            ])
-          },
-          skip: () => !yargs.restart,
-        })
-        tasks.push({
-          title: `Saving ${process} state for future startup...`,
-          task: async (_ctx, task) => {
-            await sshExec(ssh, sshOptions, task, serverConfig.path, 'yarn', [
-              'pm2',
-              'save',
-            ])
-          },
-          skip: () => !yargs.restart,
-        })
+        if (serverConfig.pm2Global) {
```

Comment:
Rather than repeating these entire commands when only a couple of things change, what if just simplify it into setting the name of the command and the options first, then run the command once but using those vars for the settings? Something like:

```javascript
// assume local install
let command = 'yarn'
let startOptions = ['pm2', 'start', 'ecosystem.config.js', '--only', process]
let restartOptions = ['pm2', 'restart', process]
let saveOptions = ['pm2', 'save']

if (serverConfig.pm2Global) {
  command = 'pm2'
  startOptions.shift()
  restartOptions.shift()
  saveOptions.shift()
}

tasks.push({
  title: `Starting ${process} process for the first time...`,
  task: async (_ctx, task) => {
    await sshExec(ssh, sshOptions, task, serverConfig.path, command, options)
  },
  skip: () => !yargs.restart,
})

// save...

// restart...
```